.github/workflows/make.yml

```diff
@@ -0,0 +1,26 @@
+name: Build and Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  run-tests:
+    name: Run tests
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
+
+      - name: Build files
+        run: make all 
+
+      - name: Run tests
+        run: make run_all_tests
```

Comment:
We will need to adapt the tests that are currently on the repository to return 1 (or an error code) if an error happens. Can you do it on this PR?

Reply:
Yes I can take care of standardizing all the tests